Revert 85d7889e9b90c3c35bc592013b942f72d5475baf to keep targeting Windows 7 by default.

From 85d7889e9b90c3c35bc592013b942f72d5475baf Mon Sep 17 00:00:00 2001
From: Andrey Semashev <andrey.semashev@gmail.com>
Date: Sun, 8 Oct 2023 20:28:42 +0300
Subject: [PATCH] Target Windows 10 by default and the latest Windows if 10 is
 not supported.

---
 boost/winapi/config.hpp | 10 --------++
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/boost/winapi/config.hpp b/boost/winapi/config.hpp
index d15e93d..aadc0ae 100644
--- a/boost/winapi/config.hpp
+++ b/boost/winapi/config.hpp
@@ -108,19 +108,13 @@
 #elif defined(WINVER)
 #define BOOST_USE_WINAPI_VERSION WINVER
 #else
+// By default use Windows 7 API on compilers that support it and Vista or XP on the others
-// By default use Windows 10 API on compilers that support it and latest versions on the others
 #if (defined(_MSC_VER) && _MSC_VER < 1500) || defined(BOOST_WINAPI_IS_MINGW)
 #define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WINXP
 #elif (defined(_MSC_VER) && _MSC_VER < 1600)
 #define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WIN6
+#else
-#elif (defined(_MSC_VER) && _MSC_VER < 1700)
 #define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WIN7
-#elif (defined(_MSC_VER) && _MSC_VER < 1800)
-#define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WIN8
-#elif (defined(_MSC_VER) && _MSC_VER < 1900)
-#define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WINBLUE
-#else
-#define BOOST_USE_WINAPI_VERSION BOOST_WINAPI_VERSION_WIN10
 #endif
 #endif
 #endif
